### DISCUSSION TITLE
检测到 geosite.dat, geoip.dat 文件或 v2ray-core 可能未正确安装，请检查

### DISCUSSION THREAD

**Discussion by YummyNovaSoup** [v2rayA/v2rayA#1453] [Q&A] [answered]
这是什么原因呢，我下载的debian installer文件本地安装的

**Comment by YummyNovaSoup** [comment, marked as answer]
已解决，步骤没有做完整。未安装v2ray-core和xray-core

**Reply by KirsminX** [reply]
> 怎么安装呢？

我解决这个问题了。按照以下办法

切换到 root 账号
`su root`
下载缺失文件
```bash
cd /etc/v2raya/
wget https://github.com/Loyalsoldier/v2ray-rules-dat/releases/latest/download/geoip.dat
wget https://github.com/Loyalsoldier/v2ray-rules-dat/releases/latest/download/geosite.dat
```
安装 V2Ray-Core。注意：**修改为你自己的架构，以下为 ARM 64 版本**
```bash
mkdir v2ray
cd v2ray
wget https://github.com/v2fly/v2ray-core/releases/download/v5.25.1/v2ray-linux-arm64-v8a.zip
unzip v2ray-linux-*
mv v2ray /usr/bin/
```
配置 V2RayA
编辑 `/etc/v2raya/config.json`，添加以下内容：
```json
{
  "v2ray_path": "/usr/bin/v2ray"
}
```
保存，重启 v2raya
`systemctl restart v2raya`
Done!!!!

**Reply by luzhixing12345** [reply]
同样的问题，不过我是因为 deb 用了 x86 的，换成 x64 的就好了

**Reply by CoolBoost** [reply]
有一个地方不对，geosite.dat和geoip.dat应该放在/usr/share/v2ray目录下，而不是/etc/v2raya

**Reply by 110LK** [reply]
真的有效！👍👍👍

**Reply by 1260zqs** [reply]
> 有一个地方不对，geosite.dat和geoip.dat应该放在/usr/share/v2ray目录下，而不是/etc/v2raya

从源代码看是 /usr/share/v2raya 目录下

**Reply by Anware-Canella** [reply]
> > 有一个地方不对，geosite.dat和geoip.dat应该放在/usr/share/v2ray目录下，而不是/etc/v2raya
> 
> 从源代码看是 /usr/share/v2raya 目录下

试过了，就是/usr/share/v2ray

**Reply by 1260zqs** [reply]
> > > 有一个地方不对，geosite.dat和geoip.dat应该放在/usr/share/v2ray目录下，而不是/etc/v2raya
> > 
> > 
> > 从源代码看是 /usr/share/v2raya 目录下
> 
> 试过了，就是/usr/share/v2ray

如果用一些预览版的v2raya路径就会变, 新版本代码改了
<img width="555" height="721" alt="image" src="https://github.com/user-attachments/assets/09b080eb-86c4-4b2e-ae07-9b876c7f4c85" />

**Reply by lrtfm** [reply]
下载源码编译的版本，出现如下日志：
```
❯ XDG_DATA_DIRS="/opt/homebrew/share:/usr/local/share:/usr/share" ./v2raya --lite
[DEBUG] main.main started
2026/06/01 01:27:05.180 [I] [sqlite.go:91]  Initializing database schema
2026/06/01 01:27:05.181 [I] [sqlite.go:91]  Database schema initialized successfully
2026/06/01 01:27:05.181 [A] [main.go:23]  downloading missing geoip.dat and geosite.dat
2026/06/01 01:27:05.182 [I] [geoip.go:17]  Downloading https://github.com/v2fly/geoip/releases/latest/download/geoip.dat to /Users/xyz/Library/Application Support/v2raya/geoip.dat.new
2026/06/01 01:27:18.204 [I] [pre.go:255]  download[geoip.dat]: SUCCESS

2026/06/01 01:27:18.205 [I] [geosite.go:18]  Downloading https://github.com/v2fly/domain-list-community/releases/latest/download/dlc.dat to /Users/xyz/Library/Application Support/v2raya/geosite.dat.new
2026/06/01 01:27:20.664 [I] [pre.go:261]  download[geosite.dat]: SUCCESS

2026/06/01 01:27:20.664 [A] [main.go:23]  geoip.dat and geosite.dat are ready
2026/06/01 01:27:20.665 [A] [main.go:25]  V2Ray binary is /Users/xyz/opt/v2raya/v2rayA/v2raya_core
2026/06/01 01:27:20.665 [A] [main.go:25]  V2Ray asset directory is /Users/xyz/Library/Application Support/v2raya
2026/06/01 01:27:20.665 [A] [main.go:25]  v2rayA working directory is /Users/xyz/opt/v2raya/v2rayA
2026/06/01 01:27:20.665 [A] [main.go:25]  v2rayA configuration directory is /Users/xyz/Library/Application Support/v2raya
```

上面显示下载了相应dat文件，但是查看目录，发现相应文件变成了指向自己的链接：
```
❯ ls -la "/Users/xyz/Library/Application Support/v2raya/"
total 4384
drwxr-x---    8 xyz  staff      256  6月  1 01:27 .
drwx------+ 195 xyz  staff     6240  5月 19 17:33 ..
-rw-------    1 xyz  staff       79  6月  1 01:14 config.json
lrwxr-xr-x    1 xyz  staff       58  6月  1 01:27 geoip.dat -> /Users/zzyang/Library/Application Support/v2raya/geoip.dat
-rw-r--r--    1 xyz  staff  2149920  6月  1 01:27 geosite.dat
-rw-r--r--    1 xyz  staff    57344  6月  1 01:19 v2raya.db
-rw-r--r--    1 xyz  staff    32768  6月  1 01:27 v2raya.db-shm
-rw-r--r--    1 xyz  staff        0  6月  1 01:27 v2raya.db-wal
```

如下启动则文件正常连接到配置的资源目录：
```
XRAY_LOCATION_ASSET=/Users/xyz/opt/v2raya/conf/dat ./v2raya --lite
```

这样看来 macos 上默认的 XRAY_LOCATION_ASSET 目录是 /Users/xyz/Library/Application Support/，导致了上面的循环引用，从而引发错误。

**Comment by zmustlivez** [comment]
Can anybody help - i didn't want install v2raya, i was download v2raya_linux_x64_2.2.6.7 on fedora, download geoip.dat, and geosite.dat to /etc/v2raya, added "v2ray_path": "_custom_path_apps/v2raya" (where "v2raya_linux_x64_2.2.6.7"), in brower at address http://127.0.0.1:2017/ opened page, saw my proxy, but have window "failed to start v2ray-core: geoip.dat or geosite.dat file does not exists", what i must do else?

**Reply by hualayn** [reply]
v2raya is just a web ui, front. the core is xray, backend. you could run the program just by xray, but the settings of xray is not for human, advanced settings even. don't install v2ray, use xray instead. geoip.dat is useless, geosite.dat is not necessary. 
so, just install v2raya & xray, start v2raya, browse 127.0.0.1:2017, input your proxy, connect and run
if iabove is alright, goto settings, set proxy mode to whitelist and save
about geosite.dat, search "loyalsoldier" for more details

**Reply by zmustlivez** [reply]
> v2raya is just a web ui, front. the core is xray, backend. you could run the program just by xray, but the settings of xray is not for human, advanced settings even. don't install v2ray, use xray instead. geoip.dat is useless, geosite.dat is not necessary. so, just install v2raya & xray, start v2raya, browse 127.0.0.1:2017, input your proxy, connect and run if iabove is alright, goto settings, set proxy mode to whitelist and save about geosite.dat, search "loyalsoldier" for more details

Hello, hualayun, thank you very much for answer. All problem in that - i don't want install v2raya, i want use "portable version" of its, like *.appimage or link *.sh with option "start like app".
I want use some port for vpn - example 2080, other ports will be bypass


I was found solution
Unzip xray and v2raya at one directory, and create link in ~ /.config/autostart/
With next contain:
Desktop Entry] Type=Application Exec=env XDG_RUNTIME_DIR=~ /.local/share V2RAYA_V2RAY_BIN=/mnt/bbc449ec-5751-4ff3-957c-b35cc7d83c90/Various/apps/v2raya/xray /mnt/bbc449ec-5751-4ff3-957c-b35cc7d83c90/Various/apps/v2raya/v2raya_linux_x64_2.2.7.1 --lite Hidden=false X-GNOME-Autostart-enabled=true Name=v2rayA Comment=Start v2rayA with assets in ~/.local/share/v2ray

**Reply by hualayn** [reply]
hello, zmustlivez! maybe v2rayN is more suitable. search v2rayN in github, there are appimage and deb packages to install directly. 
about ports, i dont know exactly what you mean(sorry for my poor english). change the default port 2017?
and last thing, look into xtls.github.io to know more about the xray project.

**Reply by zmustlivez** [reply]
> hello, zmustlivez! maybe v2rayN is more suitable. search v2rayN in github, there are appimage and deb packages to install directly. about ports, i dont know exactly what you mean(sorry for my poor english). change the default port 2017? and last thing, look into xtls.github.io to know more about the xray project.

Hi, hualayn
My english more biggest poor than you))
Some time early i was trying to use v2rayn, but have a some problem with config autostart of app, and refused of using it, and continue used nekobox, but now, when i successed config v2raya and xray (link that you write) i refused of nekobox, and nearlest time dont want experiments))
Thanks for help)

**Reply by hualayn** [reply]
hello, zmustlivez! 
will be very glad if i could help :)

**Comment by woodlyer** [comment]
最新的版本中， 搜索这些.dat 的目录是~/.local/share/v2raya   ，应该把geosite.dat, geoip.dat放到这个目录下。 
之前的版本是可以放到v2raya所在的目录下的。 
很奇怪为什么要修改。   建议把v2raya所在的目录也添加到搜索目录中。
